qemu-tests/src/interrupts.rs

```diff
@@ -0,0 +1,229 @@
+//! Interrupt code for the MPS2-AN505
```

Comment:
This sure is a lot of boilerplate - it seems like a lot of unrelated code just to bring up something in qemu.

Having said that, I don't have a better suggestion (don't know enough about bare metal on qemu etc.), so this might just be something that we need to do here